### PR TITLE
tooling: Update `envoy.docs.sphinx_runner`

### PR DESCRIPTION
**Pull request by phlax**
- Quieten docs build
 - Run sphinx multi core

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
i will need to land this one as the website needs to be updated at the same time - https://github.com/envoyproxy/envoy-website/pull/257

File: docs/conf.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def _config(key):
 sys.path.append(os.path.abspath("./_ext"))
 
 extensions = [
-    'sphinxcontrib.httpdomain', 'sphinx.ext.extlinks', 'sphinx.ext.ifconfig',
+    'envoy.docs.sphinx_runner.ext.httpdomain', 'sphinx.ext.extlinks', 'sphinx.ext.ifconfig',
```
Comment (htuch) [inline docs/conf.py:94]: What's going on here? The rest looks fine.

Comment (phlax) [reply docs/conf.py:94]: the httpdomain extension (still) doesnt work with multi-core builds (https://github.com/sphinx-contrib/httpdomain/issues/45)

im going to submit a PR to fix upstream, but in the meantime the fix is here https://github.com/envoyproxy/pytooling/blob/main/envoy.docs.sphinx_runner/envoy/docs/sphinx_runner/ext/httpdomain.py

**Review by htuch — APPROVED** [review]

**Pull request merged**